python/ray/train/tests/test_lightning_trainer.py

```diff
@@ -0,0 +1,125 @@
+import numpy as np
+from ray.train.lightning import LightningConfigBuilder, LightningTrainer
+import ray
+from ray.air.util.data_batch_conversion import convert_batch_type_to_pandas
+import pytest
+
+from ray.train.tests._lightning_utils import (
+    LinearModule,
+    DoubleLinearModule,
+    DummyDataModule,
+)
+
+
+@pytest.mark.parametrize("accelerator", ["cpu", "gpu"])
+@pytest.mark.parametrize("datasource", ["dataloader", "datamodule"])
+def test_trainer_with_native_dataloader(accelerator, datasource):
+    num_epochs = 4
+    batch_size = 8
+    num_workers = 2
+    dataset_size = 256
+
+    config_builder = LightningConfigBuilder()
+    config_builder.set_module_class(LinearModule)
+    config_builder.set_module_init_config(input_dim=32, output_dim=4)
+    config_builder.set_trainer_init_config(
+        max_epochs=num_epochs, accelerator=accelerator
+    )
+    lightning_config = config_builder.build()
+
+    datamodule = DummyDataModule(batch_size, dataset_size)
+    train_loader = datamodule.train_dataloader()
+    val_loader = datamodule.val_dataloader()
+
+    if datasource == "dataloader":
+        config_builder.set_trainer_fit_params(
+            train_dataloaders=train_loader, val_dataloaders=val_loader
+        )
+    if datasource == "datamodule":
+        config_builder.set_trainer_fit_params(datamodule=datamodule)
+
+    scaling_config = ray.air.ScalingConfig(
+        num_workers=num_workers, use_gpu=(accelerator == "gpu")
+    )
+
+    trainer = LightningTrainer(
+        lightning_config=lightning_config, scaling_config=scaling_config
+    )
+
+    trainer.fit()
+
```

Comment:
should we add an assert in the `training_step` to see if the data is being passed in as expected?

Reply:
We may not need to do that. The training_step function will fail if the input batch is not in the expected format.